app/controllers/concerns/shopify_app/ensure_has_session.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module ShopifyApp
+  module EnsureHasSession
```

Comment:
Rather than duplicating this, I think inheriting would be preferrable for maintenance. But I'm unsure which way round, `EnsureHasSession < Authenticated` or `Authenticated < EnsureHasSession`...

Reply:
Ignore, I see that's what we're doing already.